src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int ARRAY_SIZE = 6;
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[ARRAY_SIZE];
+        for (int i = 0; i < ARRAY_SIZE; i++) {
+            if (i < (ARRAY_SIZE / 2)) {
+                figures[i] = new FigureSupplier().getRandomFigure();
+            } else {
+                figures[i] = new FigureSupplier().getDefaultFigure();
```

Comment:
Don't create your FigureSupplier inside the loop. See [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#you-should-create-several-random-figures-so-you-will-use-a-loop-please-dont-create-a-new-figuresupplier-inside-the-loop).